doc/source/overview/config_files.md

```diff
@@ -0,0 +1,753 @@
+`preprocess_fn`, which themselves rely on other artefacts. TOML files are highly flexible when it comes to specifying 
+dictionaries; each section in a TOML file, demarcated by section headers enclosed in [ ] brackets, is actually a Python 
+dictionary (when the file is parsed). For example, the `preprocess_fn` can be specified as a dictionary with:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[preprocess_fn]
+src = "function.dill"
+```
+
+Alternatively, the following are also equivalent:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+preprocess_fn = {src="function.dill"}
+```
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+preprocess_fn.src = "function.dill"
+```
+
+The above schemas are not strictly necessary in this case, since `preprocess_fn` could simply be specified with
+`preprocess_fn = "function.dill"` here. However, the need to specify fields as dictionaries will become clear in the 
+following sections, where the layouts of the various possible artefact dictionaries are documented.
+
+#### Preprocessing function
+
+Simple `preprocess_fn`'s can be specified directly as a serialized dill file or via a function registry. If additional 
+arguments are to be passed at call time, the `preprocess_fn` can instead be specified as a dictionary:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[preprocess_fn]
+src = "function.dill"
+kwargs = {kwarg1="a", kwarg2=1.5, kwarg3=true}
+```
+
+In this case, the `src` field can be a locally stored `.dill` file or a function registry, and `kwargs` is a 
+dictionary of keyword arguments to be passed to the function. As noted previously, there is considerable flexibility 
+with regard to how these fields are specified. For example, the following is equivalent:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+preprocess_fn.src = "function.dill"
+
+[preprocess_fn.kwargs]
+kwarg1 = "a"
+kwarg2 = 1.5
+kwarg3 = true
+```
+
+A special (but common!) case for `preprocess_fn` is to use the Alibi Detect utility function 
+[preprocess_drift](../api/alibi_detect.cd.tensorflow.preprocess.rst) (or its pytorch equivalent). This is a utility 
+function to incorporate a user-supplied model, batch preprocessing function, and/or tokenizer into the preprocessing 
+step of a drift detector. It is pre-registered, so can be specified with 
+`src="@cd.tensorflow.preprocess.preprocess_drift"`, and the rest of the fields in `preprocess_fn` are then the 
+args/kwargs to pass to `preprocess_drift`. For example: 
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[preprocess_fn]
+src = "@cd.tensorflow.preprocess.preprocess_drift"
+batch_size = 32
+preprocess_batch_fn = "batch_fn.dill"
+```
+
+#### Models 
+
+A `model` dictionary may be used to specify a model within a `preprocess_fn`, or a model for detectors such as the
```

Comment:
Are we happy with the semantics that a stand-alone `model` dictionary might in fact refer to one that is an argument to `preprocess_fn`? I'm not sure I'm 100% comfortable with this (and this also goes beyond our tacit assumption of keeping config files API-compatible), also what if there is both a `model` and a `preprocess_fn.model`? @arnaudvl

Reply:
As it currently stands if there were a `model` this would be passed as a detector kwarg, and if there were also a `preprocess_fn.model` this would be passed as a kwarg to `preprocess_drift`, which would then be passed into the detector `preprocess_fn`. I've just modified the `test_save_classifierdrift` to test this (i.e. it saves and loads a detector with a `model` and `preprocess_fn.model`). 

I sort of think this is the most API-compatible way to do it? Since we do indeed have a kwarg named `model` in detectors and `preprocess_drift`? We could rename the field in `preprocess_fn`, but that wouldn't match the `preprocess_drift` API. 

On a related note, the general idea of the config functionality is that the artefact dictionaries are modular/portable i.e. the same `ModelConfig` can be used to define any model, regardless of whether it is passed as a detector kwarg, a `preprocess_fn` kwarg, or even to the `proj` kwarg of `DeepKernel`. The same applies for the `KernelConfig`, which might be a detector kwarg, or a kwarg of `DeepKernel`. I'm not sure right now I see a use case for making different model, kernel config's etc for different use cases? Although I could have missed some use cases...

Reply:
Oh maybe I've been misled this whole time... My impression was that if I specify e.g. `[model]` dictionary at top-level, it would automagically be inferred as actually being `[preprocess_drift.model]`? I guess from what you've said now that is not actually the case?

Reply:
p.s. just to be clear, if the above was a `ClassifierDrift` config which takes a `model` kwarg, we would have just added a field such as `model.src = "classifier_model/"`. (note that we can skip the [ ] notation and condense onto fewer lines when desired).

Reply:
Hmm maybe, the key is to distinguish between a `model` (top-level) field  and a `model` dictionary maybe? Tho it's the same thing if `model` is top-level.

Reply:
OK I'll have a think. Ideally I'd like to get rid of lots of this "artefact dictionaries" stuff from the docs page, and offload it to the api docs page. Then be more succinct about the modular "drop-in" type approach with these dicts such as `model`. However, I've hidden `schemas.py` from the API docs for now as it doesn't render very nicely due to all the inheritance in there...

Reply:
This is resolved in the new docs.